clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/TaskDefinitionCachingAgent.java

```diff
@@ -120,8 +128,8 @@ public String getAgentType() {
   }
 
   private Set<TaskDefinition> retrieveFromCache(Set<String> taskDefArns, ProviderCache providerCache) {
+    TaskDefinitionCacheClient taskDefinitionCacheClient = new TaskDefinitionCacheClient(providerCache, objectMapper);
```

Comment:
Why is this instantiated every time the caching agent runs?  I don't feel there is a need to do that every time

Reply:
There'd be circular dependency otherwise.  It's "OK" to have it like that then